plugins/autosuspend.koplugin/main.lua

```diff
@@ -169,87 +172,97 @@ function AutoSuspend:onPreventStandby()
+                timeout = 3,
+            })
+        end,
+        extra_text = T(_("Default value: %1"),
+            util.secondsToClockDuration("modern", default_value, true, true):gsub("00m$","")),
+        extra_callback = function(year, month, day, year_widget, hour_widget, min_widget)
+            hour_widget.value = is_day_hour and math.floor(default_value / (24*3600))
+                or math.floor(default_value / 3600)
+            hour_widget:update()
+            min_widget.value = is_day_hour and math.floor(default_value / 3600) % 24
+                or math.floor(default_value / 60) % 60
+            min_widget:update()
+        end,
+        keep_shown_on_apply = true,
+    })
+end
+
 function AutoSuspend:addToMainMenu(menu_items)
     menu_items.autosuspend = {
         sorting_hint = "device",
         text_func = function()
             if self.auto_suspend_timeout_seconds  then
-                local duration_format = G_reader_settings:readSetting("duration_format", "classic")
                 return T(_("Autosuspend timeout: %1"),
-                    util.secondsToClockDuration(duration_format, self.auto_suspend_timeout_seconds, true))
+                    util.secondsToClockDuration("modern", self.auto_suspend_timeout_seconds, true, true))
             else
                 return _("Autosuspend timeout")
             end
         end,
         keep_menu_open = true,
         callback = function(touchmenu_instance)
-            local InfoMessage = require("ui/widget/infomessage")
-            local SpinWidget = require("ui/widget/spinwidget")
-            local autosuspend_spin = SpinWidget:new {
-                value = self.auto_suspend_timeout_seconds / 60,
-                value_min = 5,
-                value_max = 240,
-                value_hold_step = 15,
-                ok_text = _("Set timeout"),
-                title_text = _("Timeout in minutes"),
-                callback = function(autosuspend_spin)
-                    self.auto_suspend_timeout_seconds = autosuspend_spin.value * 60
-                    G_reader_settings:saveSetting("auto_suspend_timeout_seconds", self.auto_suspend_timeout_seconds)
-                    UIManager:show(InfoMessage:new{
-                        text = T(_("The system will automatically suspend after %1 minutes of inactivity."),
-                            string.format("%.2f", self.auto_suspend_timeout_seconds / 60)),
-                        timeout = 3,
-                    })
-                    self:_unschedule()
-                    self:_start()
-                    if touchmenu_instance then touchmenu_instance:updateItems() end
-                end
-            }
-            UIManager:show(autosuspend_spin)
+             -- 60 sec (1') is the minimum and 24*3600 sec (1day) is the maximum suspend time.
+            self:setSuspendShutdownTimes(touchmenu_instance,
+                _("Timeout for autosuspend"), _("Enter Time in hours and minutes."),
```

Comment:
English Is Not Like German And Words Like Time In The Middle Should Be Written "time" :)